p2p/store.go

```diff
@@ -0,0 +1,43 @@
+package p2p
+
+import (
+	"errors"
+	"sync"
+)
+
+type SyncerStore interface {
+	AddPeer(addr string) error
+	RandomPeer() (string, error)
+}
+
+type EphemeralStore struct {
+	mu    sync.Mutex
+	peers map[string]struct{}
+}
+
+func (e *EphemeralStore) AddPeer(addr string) error {
+	e.mu.Lock()
+	e.peers[addr] = struct{}{}
+	e.mu.Unlock()
+	return nil
+}
+
+func (e *EphemeralStore) RandomPeer() (string, error) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	// map iteration order in go is random
+	for peer := range e.peers {
+		return peer, nil
+	}
+	return "", errors.New("no peers in list")
+}
+
+func NewEphemeralStore(peers []string) *EphemeralStore {
```

Comment:
I would drop the argument here. You can initialize the store by calling `AddPeer`.